AppCenterDataStorage/AppCenterDataStorage/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -53,65 +55,71 @@ - (instancetype)init {
 
 #pragma mark - Table Management
 
-- (BOOL)upsertWithPartition:(NSString *)partition
-            documentWrapper:(MSDocumentWrapper *)documentWrapper
-                  operation:(NSString *_Nullable)operation
-                    options:(MSBaseOptions *)options {
+- (BOOL)createUserStorageWithAccountId:(NSString *)accountId {
+
+  // Create table based on the schema.
+  return [self.dbStorage createTable:[NSString stringWithFormat:kMSUserDocumentTableNameFormat, accountId]
+                       columnsSchema:[MSDBDocumentStore columnsSchema]
+             uniqueColumnsConstraint:@[ kMSPartitionColumnName, kMSDocumentIdColumnName ]];
+}
+
+- (BOOL)upsertWithToken:(MSTokenResult *)token
+        documentWrapper:(MSDocumentWrapper *)documentWrapper
+              operation:(NSString *_Nullable)operation
+                options:(MSBaseOptions *)options {
+  // Compute expiration time as now + device time to live (in seconds).
+  // If device time to live is set to infinite, set expiration time as null in the database.
+  // Note: If the cache/store is meant to be disabled, this method should not even be called.
   NSDate *now = [NSDate date];
-  NSDate *expirationTime = [now dateByAddingTimeInterval:options.deviceTimeToLive];
-  NSString *tableName = [MSDBDocumentStore tableNameForPartition:partition];
-  NSString *insertQuery =
-      [NSString stringWithFormat:@"REPLACE INTO \"%@\" (\"%@\", \"%@\", \"%@\", \"%@\", \"%@\", \"%@\", \"%@\", \"%@\") "
-                                 @"VALUES ('%@', '%@', '%@', '%@', '%@', '%@', '%@', '%@')",
-                                 tableName, kMSPartitionColumnName, kMSDocumentIdColumnName, kMSDocumentColumnName, kMSETagColumnName,
-                                 kMSExpirationTimeColumnName, kMSDownloadTimeColumnName, kMSOperationTimeColumnName,
-                                 kMSPendingOperationColumnName, partition, documentWrapper.documentId, documentWrapper.jsonValue,
-                                 documentWrapper.eTag, [MSUtility dateToISO8601:expirationTime],
-                                 [MSUtility dateToISO8601:documentWrapper.lastUpdatedDate], [MSUtility dateToISO8601:now], operation];
+  NSString *isoExpirationTime;
+  if (options.deviceTimeToLive == MSDataStoreTimeToLiveInfinite) {
+    isoExpirationTime = kMSNullString;
+  } else {
+    isoExpirationTime =
+        [NSString stringWithFormat:@"\"%@\"", [MSUtility dateToISO8601:[now dateByAddingTimeInterval:options.deviceTimeToLive]]];
```

Comment:
The expiration time, per the spec and schema, must be an integer - (or `long` in objective c to prevent overflows)- so this must be changed to the number of seconds since 1970 and -1 for infinite (instead of `NULL`)

Reply:
Let's fix this later